hw/opentitan/ot_otp_dj.c

```diff
@@ -3507,22 +3572,25 @@ static void ot_otp_dj_pwr_load_tokens(OtOTPDjState *s)
 static void ot_otp_dj_pwr_initialize_partitions(OtOTPDjState *s)
 {
     for (unsigned ix = 0; ix < OTP_PART_COUNT; ix++) {
+        /* sanity check: all secret partitions are also buffered */
+        g_assert(!OtOTPPartDescs[ix].secret || OtOTPPartDescs[ix].buffered);
+
         if (ot_otp_dj_is_ecc_enabled(s) && OtOTPPartDescs[ix].integrity) {
             if (ot_otp_dj_apply_ecc(s, ix)) {
                 continue;
             }
         }
 
         if (OtOTPPartDescs[ix].sw_digest) {
-            uint64_t digest = ot_otp_dj_get_part_digest(s, (int)ix);
+            uint64_t digest = ot_otp_dj_get_part_digest(s, ix);
```

Comment:
I haven't yet looked at the keymgr e2e test CI failures, but one more issue I found: for the register digest reads to be correct for SW digests, this needs to be written to the `pctrl` digest.
```suggestion
            s->partctrls[ix].digest = ot_otp_dj_get_part_digest(s, ix);
```
(this is already done for HW digests in `ot_otp_dj_check_buffered_partition_integrity`, but nothing handles this case for SW digests).

*Edit*: The keymgr e2e tests are no longer failing, so I guess this was the issue causing those tests to fail as well.